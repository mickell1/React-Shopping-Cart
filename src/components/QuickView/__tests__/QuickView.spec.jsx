import React from 'react';
import { shallow } from 'enzyme';
import QuickView from '../QuickView';

describe('QuickView', () => {
  let wrapper;

  const defaultProps = {
    closeModal: jest.fn(),
    handleClose: jest.fn(),
    product: {
      title: 'Samsung Galaxy S7',
      price: 500,
      image: null,
    }
  }

  beforeEach(() => {
    wrapper = shallow(<QuickView {...defaultProps} /> )  
  })

  it('should render the QuickView Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
});