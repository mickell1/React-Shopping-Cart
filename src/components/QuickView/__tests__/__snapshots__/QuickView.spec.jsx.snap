// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickView should render the QuickView Component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuickView
    closeModal={[MockFunction]}
    handleClose={[MockFunction]}
    product={
      Object {
        "image": null,
        "price": 500,
        "title": "Samsung Galaxy S7",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modal"
      >
        <button
          className="close"
          onClick={[Function]}
          type="button"
        >
          ×
        </button>
        <div
          className="quick-view"
        >
          <div
            className="quick-view-image"
          >
            <img
              alt="Samsung Galaxy S7"
              src={null}
            />
          </div>
          <div
            className="quick-view-details"
          >
            <span
              className="product-name"
            >
              Samsung Galaxy S7
            </span>
            <span
              className="product-price"
            >
              500
            </span>
          </div>
        </div>
      </div>,
      "className": "modal-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="close"
            onClick={[Function]}
            type="button"
          >
            ×
          </button>,
          <div
            className="quick-view"
          >
            <div
              className="quick-view-image"
            >
              <img
                alt="Samsung Galaxy S7"
                src={null}
              />
            </div>
            <div
              className="quick-view-details"
            >
              <span
                className="product-name"
              >
                Samsung Galaxy S7
              </span>
              <span
                className="product-price"
              >
                500
              </span>
            </div>
          </div>,
        ],
        "className": "modal",
      },
      "ref": "modal",
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "×",
            "className": "close",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "×",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="quick-view-image"
              >
                <img
                  alt="Samsung Galaxy S7"
                  src={null}
                />
              </div>,
              <div
                className="quick-view-details"
              >
                <span
                  className="product-name"
                >
                  Samsung Galaxy S7
                </span>
                <span
                  className="product-price"
                >
                  500
                </span>
              </div>,
            ],
            "className": "quick-view",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Samsung Galaxy S7"
                  src={null}
                />,
                "className": "quick-view-image",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Samsung Galaxy S7",
                  "src": null,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="product-name"
                  >
                    Samsung Galaxy S7
                  </span>,
                  <span
                    className="product-price"
                  >
                    500
                  </span>,
                ],
                "className": "quick-view-details",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Samsung Galaxy S7",
                    "className": "product-name",
                  },
                  "ref": null,
                  "rendered": "Samsung Galaxy S7",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 500,
                    "className": "product-price",
                  },
                  "ref": null,
                  "rendered": 500,
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          >
            ×
          </button>
          <div
            className="quick-view"
          >
            <div
              className="quick-view-image"
            >
              <img
                alt="Samsung Galaxy S7"
                src={null}
              />
            </div>
            <div
              className="quick-view-details"
            >
              <span
                className="product-name"
              >
                Samsung Galaxy S7
              </span>
              <span
                className="product-price"
              >
                500
              </span>
            </div>
          </div>
        </div>,
        "className": "modal-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="close"
              onClick={[Function]}
              type="button"
            >
              ×
            </button>,
            <div
              className="quick-view"
            >
              <div
                className="quick-view-image"
              >
                <img
                  alt="Samsung Galaxy S7"
                  src={null}
                />
              </div>
              <div
                className="quick-view-details"
              >
                <span
                  className="product-name"
                >
                  Samsung Galaxy S7
                </span>
                <span
                  className="product-price"
                >
                  500
                </span>
              </div>
            </div>,
          ],
          "className": "modal",
        },
        "ref": "modal",
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "×",
              "className": "close",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "×",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="quick-view-image"
                >
                  <img
                    alt="Samsung Galaxy S7"
                    src={null}
                  />
                </div>,
                <div
                  className="quick-view-details"
                >
                  <span
                    className="product-name"
                  >
                    Samsung Galaxy S7
                  </span>
                  <span
                    className="product-price"
                  >
                    500
                  </span>
                </div>,
              ],
              "className": "quick-view",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Samsung Galaxy S7"
                    src={null}
                  />,
                  "className": "quick-view-image",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Samsung Galaxy S7",
                    "src": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="product-name"
                    >
                      Samsung Galaxy S7
                    </span>,
                    <span
                      className="product-price"
                    >
                      500
                    </span>,
                  ],
                  "className": "quick-view-details",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Samsung Galaxy S7",
                      "className": "product-name",
                    },
                    "ref": null,
                    "rendered": "Samsung Galaxy S7",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 500,
                      "className": "product-price",
                    },
                    "ref": null,
                    "rendered": 500,
                    "type": "span",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
